js/components/adPanel.js

```diff
@@ -74,15 +74,17 @@ var AdPanel = React.createClass({
 
     var isLive = this.props.currentAdsInfo.currentAdItem.isLive;
 
-    var remainingTime;
-    if (isLive) {
-      remainingTime = parseInt((this.props.adStartTime + this.props.adVideoDuration * 1000 - new Date().getTime())/1000);
-    } else {
-      remainingTime = parseInt(this.props.adVideoDuration - this.props.currentPlayhead)
+    if (this.props.skinConfig.adScreen.showAdCountDown) {
+      var remainingTime;
+      if (isLive) {
+        remainingTime = parseInt((this.props.adStartTime + this.props.adVideoDuration * 1000 - new Date().getTime())/1000);
+      } else {
+        remainingTime = parseInt(this.props.adVideoDuration - this.props.currentPlayhead)
+      }
+      remainingTime = Utils.formatSeconds(Math.max(0, remainingTime));
```

Comment:
we should probably have a check to make sure we have a finite number. Then we can avoid possible showing NaN on the screen.